### ISSUE TITLE
针对GeyserExtras 1.21.0-v1.1.4的中文更新

### ISSUE THREAD

**Issue by MengZe2l** [GeyserExtras/GeyserExtras#10]
## 针对GeyserExtras 1.21.0-v1.1.4的中文更新

汉化了绝大部分内容，可以正常开服
![20240914100105-e08524a6da34ed679ed98311dd23ab49.jpg](https://s2.loli.net/2024/09/14/3Hng1TDwjfJEvQu.jpg)
![20240914100105-79163d15ee8c236daf9b1eefe9846370.jpg](https://s2.loli.net/2024/09/14/XimNnEfCMYOwHth.jpg)
![20240914100105-57a02de3eb02d5c9b58f149beb301920.jpg](https://s2.loli.net/2024/09/14/HEJRI7inrvTuUtG.jpg)

会持续更新，看：
https://github.com/MengZe2l/GeyserExtras_CN/

作者赶紧合并(怒)

**Comment by letsgoawaydev** [comment]
Do you want to help with translations for GeyserExtras v2.0? I'm planning to add support for other locales/languages soon

**Comment by MengZe2l** [comment]
Of course, please send me the copy of GeyserExtras v2.0 after development, and I can translate it for you. We have two people developing together, and the other one is @github.com/ning-g-mo

**Comment by letsgoawaydev** [comment]
If you would like to you can join my discord here, where other translators for other languges are and I can let you know about any updates for when it comes out: https://discord.gg/QcjC5Ez2Ac

**Comment by MengZe2l** [comment]
But, I'm sorry, I'm from China, you know, where network management is very strict, and I can't download Discord using VPN, but I can tell you my email address. It's i@mengze2.cn. You can contact me by email (I like that you can recommend me to enter your Github organization as a translator, and then release the internationalization version directly or do internationalization directly in the plug-in)

**Comment by MengZe2l** [comment]
I hope you are developing version 2.0, including the lang.Java file or lang.yml file. Please provide a basic template for the English language file, and we will handle the rest of the internationalization (i18n) work. We will fork your repository, translate the necessary files, and submit a pull request after completion. You can then merge the changes into your repository

**Comment by letsgoawaydev** [comment]
I'll let you know when locale support is added, but Im not able to work on GeyserExtras for a few weeks so it might take a while, ill let you know about any updates here

**Comment by MengZe2l** [comment]
> I'll let you know when locale support is added, but Im not able to work on GeyserExtras for a few weeks so it might take a while, ill let you know about any updates here

ok, I'll read the information here often, and when it is updated, I will do the translation work

**Comment by letsgoawaydev** [comment]
im alright with people reuploading it if it adds languages but i dont support it as in im not making it so proceed at your own risk

**Comment by letsgoawaydev** [comment]
as long as people dont impersonate me / make it clear that its not by me and also not make virus jars im alright with that

**Comment by MengZe2l** [comment]
> https://hangar.papermc.io/Guardian1l/GeyserExtras_CN
> 
> Curious how it was reuploaded, not against License from what I can see but still, curious

Well, we just added multiple languages for the plug-in, no business, and we shouldn't violate the license

**Comment by MengZe2l** [comment]
> https://hangar.papermc.io/Guardian1l/GeyserExtras_CN
> 
> Curious how it was reuploaded, not against License from what I can see but still, curious

MIT has a "including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software "

**Comment by letsgoawaydev** [comment]
Thanks for crediting me on the modrinth pages description too, really appreciate it. 

Locale support is probably going to be the form of either a JSON or .lang file like how bedrock does their languages

Like I said it will be a few weeks until I can work on it again but when its done I'll let you know

**Comment by letsgoawaydev** [comment]
Hi, sorry for the long wait, but the latest build now supports languages. Here is a quick guide on how to make one

You dont have to rush with the translations: they will automatically download the latest ones from [this repository](https://github.com/GeyserExtras/data) when GeyserExtras is booted up.

All languages will be downloaded to `GeyserExtras/cache/langs/`.

To get started with editing, make sure Geyser/GeyserExtras is closed and then add the language you are translating from here:
https://github.com/GeyserExtras/data/blob/main/langs/language_names_all.json
to `GeyserExtras/cache/langs/language_names.json`.

I will be using French (`fr_FR`) as an example.

Original `language_names.json`:
```json
[
    [ "en_US", "English (United States)" ],
    [ "en_GB", "English (United Kingdom)" ]
]
```
Edited for French:
```json
[
    [ "en_US", "English (United States)" ],
    [ "en_GB", "English (United Kingdom)" ],
    [ "fr_FR", "Français (France)" ]
]
```

Then, make a json file based on your language code. You can duplicate the [`GeyserExtras/cache/langs/en_US.json`](https://github.com/GeyserExtras/data/blob/main/langs/en_US.json) if you would like, for French you would make a `GeyserExtras/cache/langs/fr_FR.json`.


Make your edits in this file, and then boot up GeyserMC and Minecraft Bedrock, ensuring on Bedrock that your language is set to the same one you are editing/adding to GeyserMC. Then check the menus and your translations should be applied. Once you are done you can make a pull request on the [repository](https://github.com/GeyserExtras/data).

**Comment by MengZe2l** [comment]
OK, we have started the translation work, but I found that you still have some places that are not multilingual. I'll give you a list of words that aren't multilingual later

**Comment by MengZe2l** [comment]
OK, we have started the translation work, but I found that you still have some places that are not multilingual. I'll give you a list of words that aren't multilingual later

**Comment by CJYKK** [comment]
Hi, I also want to contribute to the Simplified Chinese translation, but I couldn't find the language files in the repository until I saw this Issue. Is there a plan to include the address of the data repository in the README or somewhere else?

**Comment by letsgoawaydev** [comment]
> Hi, I also want to contribute to the Simplified Chinese translation, but I couldn't find the language files in the repository until I saw this Issue. Is there a plan to include the address of the data repository in the README or somewhere else?

All of GeyserExtra's languages are stored at https://github.com/GeyserExtras/data/
